core/src/main/scala/dagr/core/tasksystem/Pipeline.scala

```diff
@@ -78,9 +80,15 @@ abstract class Pipeline(val outputDirectory: Option[Path] = None,
 
   /** Recursively navigates dependencies, starting from the supplied task, and add all children to this.tasks. */
   private def addChildren(task : Task) : Unit = {
-    task.tasksDependingOnThisTask.filterNot(tasks.contains).foreach { child =>
-      tasks += child
-      addChildren(child)
+    // 1. find all tasks connected to this task
+    val toVisit: mutable.Set[Task] = mutable.HashSet[Task](task)
```

Comment:
Maybe also add a note here that this is to avoid stack overflows when there are really deep dependency graphs?